### PR TITLE
Cli improvements

### PR DESCRIPTION
**Pull request by DieMyst**

### CODE REVIEW

File: client/src/main/scala/fluence/client/ClientApp.scala [hunk 1]
```diff
@@ -50,6 +49,10 @@ object ClientApp extends App with slogging.LazyLogging {
   val algo = Ecdsa.signAlgo
   import algo.checker
 
+  //for terminal improvements: history, navigation
+  val terminal = TerminalBuilder.terminal()
```
Comment (alari) [inline client/src/main/scala/fluence/client/ClientApp.scala]: What happens if client app is runned in non-interactive mode? e.g. inside docker, or via supervisorctl? Could these lines throw exceptions or block in some cases?

File: client/src/main/scala/fluence/client/ClientApp.scala [hunk 2]
```diff
@@ -58,12 +61,12 @@ object ClientApp extends App with slogging.LazyLogging {
         case None             ⇒ ConfigFactory.load()
       }
 
-      val seedsB64 = if (c.seed.isEmpty) config.getStringList("fluence.seed").asScala else c.seed
+      val seedsB64 = if (c.seed.isEmpty) config.getStringList("fluence.client.seed").asScala else c.seed
```
Comment (alari) [inline client/src/main/scala/fluence/client/ClientApp.scala]: Seed nodes are shared between client and server. No need to put seeds into client or server.

File: client/src/main/scala/fluence/client/ClientApp.scala [hunk 3]
```diff
@@ -72,63 +75,61 @@ object ClientApp extends App with slogging.LazyLogging {
...
 
-  /**
-   * @param line Input string from user
-   * @param fluenceClient Ready to work client
-   * @param ac Keypair of user
-   */
-  def handleCommands(line: String, fluenceClient: FluenceClient, ac: AuthorizedClient): Unit = {
-    val commandOp = CommandParser.parseCommand(line)
-    commandOp match {
+  def handleCmds(fluenceClient: FluenceClient, ac: AuthorizedClient): Task[Unit] = {
```
Comment (alari) [inline client/src/main/scala/fluence/client/ClientApp.scala]: What is the shared resource here? Why Task is used, not IO?

File: client/src/main/scala/fluence/client/ClientApp.scala [hunk 4]
```diff
@@ -72,63 +75,61 @@ object ClientApp extends App with slogging.LazyLogging {
...
-
       case Some(Get(k)) ⇒
-        val t = for {
+        for {
           ds ← fluenceClient.getOrCreateDataset(ac)
           res ← ds.get(k)
-        } yield res
+          printRes = res match {
+            case Some(r) ⇒ r
+            case None    ⇒ "None"
```
Comment (alari) [inline client/src/main/scala/fluence/client/ClientApp.scala]: Probably it should be something more specific. "None" looks exactly like a value.

File: client/src/main/scala/fluence/client/ClientApp.scala [hunk 5]
```diff
@@ -72,63 +75,61 @@ object ClientApp extends App with slogging.LazyLogging {
...
+          }
+          _ = logger.info("Result: \"" + printRes + "\"")
+          _ ← handle
+        } yield ()
+      case _ ⇒
+        Task.pure(logger.info("Wrong command.")).flatMap(_ ⇒ handle)
```
Comment (alari) [inline client/src/main/scala/fluence/client/ClientApp.scala]: `.pure` is eager, while logging is an effect. You should use smth like `suspend` -- e.g. `IO(logger...)` or `Task(logger...)`

File: client/src/main/scala/fluence/client/ClientApp.scala [hunk 6]
```diff
@@ -72,63 +75,61 @@ object ClientApp extends App with slogging.LazyLogging {
...
+    }.asyncBoundary
```
Comment (alari) [inline client/src/main/scala/fluence/client/ClientApp.scala]: Why not to `shift` here? Why do you need an async boundary?

File: client/src/main/scala/fluence/client/ClientApp.scala (hunk 1 above)
Comment (DieMyst) [reply client/src/main/scala/fluence/client/ClientApp.scala]: I don't no for sure. We should write some bash script for connecting to cli process for this cases.

File: client/src/main/scala/fluence/client/ClientApp.scala (hunk 4 above)
Comment (DieMyst) [reply client/src/main/scala/fluence/client/ClientApp.scala]: something like <null> without quotas

**Pull request merged**